### PR TITLE
[RFC] Enable -Wconversion in tag.c

### PR DESCRIPTION
**Pull request by whwilder**
Work for #567 

This is my first contribution to a large open-source project, so if I've done anything stupid, please let me know.

### CODE REVIEW

File: src/nvim/tag.c [hunk 1]
```diff
@@ -2711,7 +2707,7 @@ expand_tags (
       tagnm[c++] = (t_p.tagkind != NULL && *t_p.tagkind)
                    ? *t_p.tagkind : 'f';
       tagnm[c++] = 0;
-      memmove((*file)[i] + c, t_p.fname, t_p.fname_end - t_p.fname);
+      memmove((*file)[i] + c, t_p.fname, (size_t)(t_p.fname_end - t_p.fname));
```
Comment (justinmk) [inline src/nvim/tag.c]: Is it certain that the result of the subtraction will be non-negative? If not then some assert or other check needs to come before this cast.

**Comment by justinmk** [comment]
Thanks! In general, search existing PRs before starting new work. Check https://github.com/neovim/neovim/pull/4031 for related discussion. Your PR is smaller so it might be easier, but I didn't look closely.

**Comment by jamessan** [comment]
Also, feel free to rebase your PR instead of merging from master.

**Comment by whwilder** [comment]
@justinmk My apologies!  I thought I had checked for other PRs, but clearly not thoroughly enough.

The other PR fixes linting errors (which seem to make up the bulk of the changes), which mine doesn't and would end up having to do anyway to pass the automated tests.  Mine also has the same issues WRT changing lengths to size_t that you mention over there.

It seems like it would be simpler to go with #4031 than with this one; I would have to do a lot of the work they've already done and my PR would probably end up being around the same size.

**Comment by justinmk** [comment]
Ok

**Pull request closed without merging**